tests/unit/widget/widget_classes.js

```diff
@@ -0,0 +1,129 @@
+		}
+	}
+});
+
+$.widget( "ui.classesWidget", {
+	options: {
+		classes: {
+			"test-wrapper": "self-wrapper",
+			"test-self": "self-class self-class-2"
+		}
+	},
+	_create: function() {
+		this.span = $( "<span>" )
+			.appendTo( this.element );
+
+		this.element.wrap( "<div>" );
+
+		this.wrapper = this.element.parent();
+
+		this._addClass( "test-self", "test-self-extra" )
+			._addClass( "test-self-2" )
+			._addClass( null, "test-self-extra-null" )
+			._addClass( this.span, null, "test-span-extra-null" )
+			._addClass( this.span, "test-span", "test-span-extra" )
+			._addClass( this.wrapper, "test-wrapper" );
+
+	},
+	toggleClasses: function( bool ) {
+		this._toggleClass( "test-self", "test-self-extra", bool )
+			._toggleClass( "test-self-2", null, bool )
+			._toggleClass( null, "test-self-extra-null", bool )
+			._toggleClass( this.span, null, "test-span-extra-null", bool )
+			._toggleClass( this.span, "test-span", "test-span-extra", bool )
+			._toggleClass( this.wrapper, "test-wrapper", null, bool );
+	},
+	removeClasses: function() {
+		this._removeClass( "test-self", "test-self-extra" )
+			._removeClass( "test-self-2" )
+			._removeClass( null, "test-self-extra-null" )
+			._removeClass( this.span, null, "test-span-extra-null" )
+			._removeClass( this.span, "test-span", "test-span-extra" )
+			._removeClass( this.wrapper, "test-wrapper" );
+	},
+	_destroy: function() {
+		this.span.remove();
+		this.element.unwrap();
+	},
+	widget: function() {
+		return this.wrapper;
+	}
+});
+
+test( ".option() - classes setter", function(){
+	expect( 12 );
+
+	var testWidget = $.ui.classesWidget(),
+		currentWrapperClass = testWidget.option( "classes.test-wrapper" );
+
+	testElementClasses( testWidget.element, true, "add" );
+
+	testWidget.option({
+		classes: {
+			"test-span": "self-span-new",
+			"test-wrapper": currentWrapperClass + " self-wrapper-new",
+			"test-self": "self-class-2"
+		}
+	});
+
+	equal( testWidget.element.is( ".test-self.self-class-2" ), true,
+		"Removing a class leaves the structure and other classes in value" );
+	equal( !testWidget.element.is( ".self-class" ), true,
+		"Removing a class from the value removes the class" );
+	testWidget.option( "classes.test-self", "" );
+	equal( testWidget.element.is( ".test-self" ), true,
+		"Setting to empty value leaves structure class" );
+	equal( !testWidget.element.is( ".self-class-2" ), true,
+		"Setting empty value removes previous value classes" );
+	equal( testWidget.span.is( ".test-span.self-span-new" ), true,
+		"Adding a class to an empty value works as expected" );
+	equal( testWidget.wrapper.is( ".test-wrapper.self-wrapper-new" ), true,
+		"Appending a class to the current value works as expected" );
+});
+
+test( ".destroy() - class removal", function(){
+	expect( 1 );
+
+	domEqual( "#widget", function(){
+		$( "#widget" ).classesWidget().classesWidget( "destroy" );
+	});
+});
+
+function testElementClasses( widget, bool, method, toggle ) {
```

Comment:
This is a bad abstraction. Custom assertions are good, but this just makes it difficult to figure out what's being tested. An optional 4th argument makes it even harder - as far as I can tell, its not even necessary, since you could also look at the `bool` and `method` arguments to calculate its value:

```js
var toggle = method === "toggle" ? ( ", " + bool ) : "";
```